internal/clients/clients.go

```diff
@@ -247,8 +248,11 @@ func (c *Cache) Get(cr auth.Credentials, o ...GetOption) (client.Client, error)
 	c.mx.RUnlock()
 
 	if ok {
-		log.Debug("Used existing client")
-		return sn, nil
+		log.Debug("Used existing client",
+			"new-expiry", time.Now().Add(c.expiry),
+		)
+		sn.expiration.Reset(c.expiry)
+		return sn.client, nil
```

Comment:
return session client directly, client expiration is extended when a client is requested, rather than used. This simplifies code as we don't have to wrap the client anymore and still preserves cached clients since their expiration is extended from each resolver that needs to use them. extending client's expiration during operations would only be useful if the expiration duration can be lower than a duration if any one client operation, which a very unlikely situation.